articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+```
+
+##### Selecting Elements
+The final concept to be covered under matrices is the selection of elements. Matrices in R work like 2-D arrays in other programming languages. Therefore all the slicing operations stand true in R. Let us assume we have a `4X4` dimensional matrix. The first quarter is a `2X2` matrix. To get the first quarter of the matrix, we will use the following code:
+
+```r
+x <- matrix(1:16, brow=TRUE, nrow=4) # matrix definition
+x[1:2,1:2] # selection operation
+```
+The output is given as follows:
+```txt
+x
+     [,1] [,2] [,3] [,4]
+[1,]    1    2    3    4
+[2,]    5    6    7    8
+[3,]    9   10   11   12
+[4,]   13   14   15   16
+
+ x[1:2,1:2]
+     [,1] [,2]
+[1,]    1    2
+[2,]    5    6
+```
+
+With the solid knowledge of vectors, matrices, and various data types, let's shift gears and move towards factors and lists.
+
+#### Factors
+In the case of supervised learning algorithms, we use factors. [classification](https://www.section.io/engineering-education/supervised-learning-algorithms/) algorithms have discrete output variables, whereas regression algorithms have continuous output variables. We can use vectors and matrices to store continuous output variables. But in the case of categorical values, we use factors.
+
+R provides us with the function factor to deal with categorical values. Examples are the best way to understand the implementation.
+
+```r
+factor_vector <- c("Big","Small","Medium")
+factored_vector <- factor(factor_vector)
+factored_vector
+```
+The above code outputs the following:
+```txt
+[1] Big Small Medium  
+Levels: Big Medium Small
+```
+You might be wondering about the levels parameter present in the output. The levels signify the relative ordering between the categorical values. In the given example, all are equal by default. Let us say we want to specify an order for the categories: Big to be 2, medium to be 1, and small to be 0. We update the code as follows:
+
+```r
+factor_vector <- c("Big", "Small","Medium")
+factored_vector <- factor(factor_vector, order=TRUE, levels=c("Small","Medium","Big"))
+factored_vector
+```
+We obtain the desired output.
+```txt
+[1] Big Small Medium
+Levels: Small < Medium < Big
+```
+The final function one must know under factors, which is used quite frequently, is `summary().` The summary function summarizes factors and gives the total number of occurrences of a particular category in the initial `factor_vector.`
+
+```r
+factor_vector <- c("Big", "Small","Medium")
+factored_vector <- factor(factor_vector, order=TRUE, levels=c("Small","Medium","Big"))
+factored_vector
+summary(factored_vector)
+```
+The following code outputs:
+```txt
+Small Medium Big
+    1     1     1
+```
+
+#### Lists
+
+Lists are data structures similar to structures in C. They can hold elements belonging to various data types. Lists are the data structure that completes the R-ecosystem by providing many features. Let us look at lists and the functions it offers.
```

Comment:
Rather than providing a definition through analogy, provide standalone definition for R uses assuming no prior knowledge. Lists are objects that can contain elements of different types - name which.